src-tauri/src/domain/resource.rs

```diff
@@ -0,0 +1,75 @@
+//! Defines resources tracked in the trace domain and implements storage.
+
+use super::storable::Storable;
+use crate::domain::{duration::Duration, has_app_name::HasAppName};
+use crate::error::Error as TraceError;
+use crate::mappers::read_file;
+use async_trait::async_trait;
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+
+/// Possible lifecycle states of a traced resource.
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub enum ResourceStatus {
+    /// Resource is fully initialized and ready.
+    Ready,
+    /// Resource has been dropped or cleaned up.
+    Dropped,
+}
+
+/// A tracked resource in the application trace.
+#[derive(Clone, Serialize, Deserialize, Debug)]
+pub(crate) struct Resource {
+    /// Optional application name this resource belongs to.
+    pub app_name: Option<String>,
+    /// Optional kind/type string of the resource.
+    pub resource_type: Option<String>,
+    /// Unique numeric ID of the resource.
+    pub id: u64,
+    /// Current state of the resource.
+    pub status: ResourceStatus,
+    /// Optional target or descriptor.
+    pub target: Option<String>,
+    /// Optional lifetime duration of the resource.
+    pub duration: Option<Duration>,
+    /// Optional source‐code location.
+    pub location: Option<String>,
+    /// Optional extra attributes in JSON or plain text.
+    pub attributes: Option<String>,
+}
+
+impl Resource {
+    /// Returns a string identifier combining `app_name` and `id`.
+    ///
+    /// # Panics
+    ///
+    /// Panics if `app_name` is `None`.
+    pub fn id(&self) -> String {
+        format!("{}.{}", self.app_name.as_ref().unwrap(), self.id)
```

Comment:
Regula de aur: fara unwrap-uri - peste tot. In momentul in care facem un unwrap peste ceva gen None sau un Err(), atunci aplicatia va da panic. Ce e si mai rau in cazul nostru e ca daca suntem intr-un thread, aplicatia principala va merge mai departe, dar functionalitatea thread-ului e pierduta.

Moduri de management pentru None-uri si Err-uri:
1. Faci un match si ai ceva comportament pentru eroare -> cel putin logging de eroare cu ce anume s-a intamplat si, daca se poate, intors o eroare catre functia care apeleaza
2. unwrap_or_default -> daca nu am nicio valoare folosesc ceva default, de exemplu 0 pentru u32, data de astazi, etc.

Deseori este preferabil sa avem valori aiurea, care sa ne lase in niste stari din care aplicatia este "recoverable" decat sa crapam un thread